docs/mgmt/readme.md

```diff
@@ -0,0 +1,658 @@
+
+An operation set is marked as a resource, only when the following conditions meet:
+
+1. The request path has even segments starting from the last `providers` segment. For instance, `/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/availabilitySets/{name}` meets this requirement, but `/{scope}/providers/Microsoft.Authorization/policyAssignments` does not.
+1. The operation set must have a GET request.
+1. The schema of `200` response of the GET request is compatible to ARM's resource definition - it must have `id`, `type` and `name` (all of them are strings).
+
+If all the above conditions are met, this operation set will be marked as a "resource" by management generator, and the schema of the `200` response of the GET request will be its corresponding `ResourceData`.
+
+There are multiple ways to tweak the criteria of identifying resources, please see [change the criteria of ARM resources](#change-the-criteria-of-arm-resources) and [change resource data](#change-resource-data) section.
+
+The name of a resource usually is derived from the name of the resource data, but with some exceptions. Please see [change resource name](#change-resource-name) for more details.
+
+If a request path is incorrectly marked as a resource, you could use a configuration to make it non-resource again. Please see [mark a request path is non-resource](#mark-a-request-path-is-a-non-resource) section.
+
+## How does the generator build hierarchical structure of resources
+
+After the management generator goes through all the resources, it is ready to build the hierarchy of resources.
+
+Despite the package `Azure.ResourceManager` is also mostly generated by the same generator, the generator has 5 built-in resources as parent fallbacks when the parent of a resource is not in this RP (The most common case of this is the `ResourceGroup` which is a parent of a lot of resources):
+
+1. `Azure.ResourceManager.Tenant`
+1. `Azure.ResourceManager.Subscription`
+1. `Azure.ResourceManager.ResourceGroup`
+1. `Azure.ResourceManager.ManagementGroup`
+1. `Azure.ResourceManager.ArmResource`
+
+For each operation set, the management generator will try to find its parent, by going through all the existing resources to see if the request path of another resource is the prefix of this operation set. After going through all of the resources, we should get a list of candidates of resources that can be the parent or grand parent of this operation set, it will take the longest path as its direct parent. If none found, the generator will try the 5 built-in resources in this order:
+1. `ResourceGroup`
```

Comment:
Is there any reason behind this order?

Reply:
It is ordered by their scopes, in ascending order. resource group is smallest in scope, tenant is biggest